### PR TITLE
add support for multi-subscriber and yaml config

### PR DESCRIPTION
**Pull request by henryzhx8**
To solve issue #223.

### CODE REVIEW

File: test/engine/controller/logtail_control.go [hunk 1]
```diff
@@ -36,6 +40,31 @@ func (l *LogtailController) Init(parent *CancelChain, fullCfg *config.Case) erro
 	// todo currently only support one fixed config with cgo
 	if len(fullCfg.Ilogtail.Config) == 1 {
 		cfg := fullCfg.Ilogtail.Config[0]
+		if len(cfg.Detail) != 0 {
+			_ = os.Remove(config.ConfigYamlFileDir)
+			_ = os.Mkdir(config.ConfigYamlFileDir, 0750)
```
Comment (EvanLjp) [inline test/engine/controller/logtail_control.go:45]: don't ignore exception when init

File: test/engine/controller/logtail_control.go [hunk 2]
```diff
@@ -36,6 +40,31 @@ func (l *LogtailController) Init(parent *CancelChain, fullCfg *config.Case) erro
...
+				if _, ok := detail["inputs"]; !ok {
+					return fmt.Errorf("lack of input plugin in the %d config detail under name %s", idx, cfg.Name)
+				}
+				if _, ok := detail["outputs"]; !ok {
+					detail["flushers"] = []map[string]interface{}{
+						{
+							"Type":    "flusher_grpc",
+							"Address": "host.docker.internal:8000",
+						},
+					}
```
Comment (EvanLjp) [inline test/engine/controller/logtail_control.go:57]: currently, only support yaml in pure plugin mode?

File: test/engine/controller/validator_control.go [hunk 3]
```diff
@@ -94,14 +94,14 @@ func (c *ValidatorController) Start() error {
...
-			projectMatch, staticMatch := staticLogCheck(group.Logs[0])
+		for group := range defaultSubscriberChan {
+			_, staticMatch := staticLogCheck(group.Logs[0])
 
 			switch {
 			case staticMatch:
-				if !projectMatch {
-					continue
-				}
+				// if !projectMatch {
```
Comment (EvanLjp) [inline test/engine/controller/validator_control.go:102]: Maybe these lines should be deleted.

File: test/config/config.go [hunk 4]
```diff
@@ -89,13 +90,13 @@ func GetDefaultCase() *Case {
...
 			Interval: "10s",
 		},
-		Subscriber: Subscriber{
-			Name: "grpc",
-			Config: map[string]interface{}{
-				"address": ":8000",
-				"network": "tcp",
-			},
-		},
+		// Subscriber: Subscriber{
```
Comment (EvanLjp) [inline test/config/config.go:93]: If don't need, please delete these lines

File: test/case/behavior/input_mocklog/ilogtail-e2e.yaml [hunk 5]
```diff
@@ -17,6 +17,16 @@ boot:
...
+            - Type: metric_mock
+              IntervalMs: 1000
+              Tags:
+                tag1: aaaa
+                tag2: bbb
+              Fields:
+                content: xxxxxx
+                time: 2017.09.12 20:55:36
       content:
         - '{"inputs":[{"type":"metric_mock","detail":{"IntervalMs": 1000,"Tags":{"tag1":"aaaa","tag2":"bbb"},"Fields":{"content":"xxxxx","time":"2017.09.12 20:55:36"}}}]}'
```
Comment (EvanLjp) [inline test/case/behavior/input_mocklog/ilogtail-e2e.yaml:31]: remove the history configuration.
 Same with the other configurations.

**Review by EvanLjp — CHANGES_REQUESTED** [review]

File: test/engine/controller/logtail_control.go (hunk 2 above)
Comment (henryzhx8) [reply test/engine/controller/logtail_control.go:57]: yaml is now available for both file and plugin mode, see #210

**Review by EvanLjp — APPROVED** [review]

**Pull request merged**